helm-v3/005-test.md

```diff
@@ -0,0 +1,27 @@
+# Test Command
+
+## Overview
+
+`helm test` gives a chart author the ability to define actions on a helm release which validate that the chart is running as intended. This is useful for ensuring that a `helm install` or `helm upgrade` created/modified Kubernetes resources successfully and that the consequential `helm release` is functional.
+* A **test** is defined as a command and some arguments executed from a container running in a Kubernetes Pod. If the command completes inside the container with an exit code of 0, it is successful. All other exit codes indicate the test failed.
+* Tests are defined in Manifests that live inside of the `templates/tests/` directory inside of a chart.
```

Comment:
This is a new requirement AFAICT which means you need to both set annotations and get the dir strutcture right.

Could we simplify and make the configuration via structure only?

```
templates/test/success
templates/test/failure
```

Or something like that? Would like to make sure there are as few things as possible to coordinate.

Reply:
That's possible and I understand the concern. 

I think the question is how do people want to organize their tests? I made the assumption that folks would write tests around a particular component in one file regardless of success or failure the way we do when writing tests while programming in go or ruby. 

I did however take a look at what folks are doing currently in the charts repo and it looks like the overwhelming majority create a single yaml file that contains a `test-success` hook on a test pod. I didn't even find one that uses the `test-failure` hook. Given that is the current state, I'm comfortable changing the proposal to align on the directory structure.

Thanks for the suggestion. I'll update.